app/components/skill-list-item.js

```diff
@@ -4,6 +4,19 @@ export default Ember.Component.extend({
   classNameBindings: ['matched'],
   tagName: 'li',
 
+  matched: Ember.computed.notEmpty('userSkill'),
+  userSkills: Ember.inject.service(),
+  user: Ember.computed.alias('currentUser.user'),
+  usersUserSkills: Ember.computed.alias('user.userSkills'),
+
+  userSkill: Ember.computed('skill', 'userSkills.userSkills.isFulfilled', function() {
+    let skill = this.get('skill');
+    let userSkills = this.get('userSkills');
+    let result = userSkills.findUserSkill(skill);
+    skill.set('matched', result);
+    return result;
```

Comment:
Only other thing I could think of is to provide a callback to the sorting computed in the parent component. It's less weird, but it's also more work "at once", I think. Not sure which one would be quicker, though.